source/_docs/installation/synology.markdown

```diff
@@ -2,78 +2,177 @@
+Edit "*~/spksrc/spk/python3/src/requirements.txt*" and add the following text to the end of the file:
 ```
+## It may happen you want to add a component to Home Assistant, but it fails to install a Python package.
+## In the logs you see a error "Unable to install package *Module_Name*", if you install that
+# package with pip, some of its required packages fail with error "Failed building wheel for *Module_Name*"
+## Add these Python modules/packages that fail with that "building wheel" error in to the list below.
 
-Use PIP to install Homeassistant package 0.64.3
+# Example format of module:
+#pythonmodule==version
 
-```bash
-# ./python3 -m pip install homeassistant==0.64.3
+## Cross compilation requirements for installing Home Assistant (Tested to work on 84.3).
+cffi==1.11.5
+bcrypt==3.1.4
+cryptography==2.3.1
+
+## Cross compilation requirement for installing "Warrant" module (Needed by "Cloud" Component)
+pycryptodome==3.7.2
+
+## Cross compilation requirement for installing "HAP_python" module (Needed by "Homekit" Component)
+curve25519-donna==1.3
+ed25519==1.4
+```
+
+Add these edits to fix OpenSSL errors when using the "[Xiaomi_Aqara](/components/Xiaomi_Aqara/)" component.<br/>
+Edit "*~/spksrc/spk/python3/Makefile*" and add the following text above the text "**include ../../mk/spksrc.spk.mk**":
+```makefile
+## Makefile variable needed to fix "_openssl.so: undefined symbol: pthread_atfork" error when using "xiaomi_aqara" component.
+export CFLAGS=-pthread
 ```
 
-Create homeassistant config directory & switch to it
+#### {% linkable_title Compiling the Python 3 package %}
 
+Modify the `docker run` command so "**XXXX**" in "*arch-**XXXX***" contains the appropriate architecture of your Synology NAS. For a list of architectures, look at this [list of architectures](https://github.com/SynoCommunity/spksrc/wiki/Architecture-per-Synology-model) accepted by [spksrc](https://github.com/SynoCommunity/spksrc).<br/>
+Depending on your computer, compilation may take a hour or more (Significantly less if you have a SSD and a moderately good CPU):
 ```bash
-# mkdir /volume1/homeassistant
-# chown homeassistant /volume1/homeassistant 
-# chmod 755 /volume1/homeassistant
-# cd /volume1/homeassistant
+# sudo docker run -it -v ~/spksrc:/spksrc -w /spksrc synocommunity/spksrc bash -c 'make clean && make setup && make -C spk/python3 arch-XXXX'
 ```
-Hint: alternatively you can also create a "Shared Folder" via Synology WebUI (e.g., via "File Station") - this has the advantage that the folder is visible via "File Station".
+After the compilation is done, you can find the Python 3 package at "*~/spksrc/packages/python3_XXXX.spk*". It should be named something like "python3_armada370-6.1_3.5.5-7.spk" with a possibly different architecture and version.
```

Comment:
After the compilation is done, you can find the Python 3 package at `~/spksrc/packages/python3_XXXX.spk`. It should be named something like `python3_armada370-6.1_3.5.5-7.spk` with a possibly different architecture and version